src/radix-tree/radix.ts

```diff
@@ -0,0 +1,235 @@
+import { create as createNode, Node, updateParent, removeDocument } from "./node";
+import { boundedLevenshtein } from "../levenshtein";
+import { getOwnProperty } from "../utils";
+
+export type Nodes = Record<string, Node>;
+
+export type FindParams = {
+  term: string;
+  exact?: boolean;
+  tolerance?: number;
+};
+
+export type FindResult = Record<string, string[]>;
+
+export function insert(root: Node, word: string, docId: string) {
+  for (let i = 0; i < word.length; i++) {
+    const currentCharacter = word[i];
+    root.docs.push(docId);
+
+    if (currentCharacter in root.children) {
+      const edgeLabel = root.children[currentCharacter].word;
+      const commonPrefix = getCommonPrefix(edgeLabel, word.substring(i));
+
+      if (edgeLabel === word.substring(i)) {
+        root.children[currentCharacter].docs.push(docId);
+        root.children[currentCharacter].end = true;
+        return;
+      }
+
+      if (commonPrefix.length < edgeLabel.length && commonPrefix.length === word.substring(i).length) {
+        const newNode = createNode(true);
+        newNode.word = word.substring(i);
+
+        newNode.children[edgeLabel[commonPrefix.length]] = root.children[currentCharacter];
```

Comment:
I guess you can also extract `newNode.children[edgeLabel[commonPrefix.length]]` into a variable here